packages/dosespot-react/src/DoseSpotMedicationSelect.tsx

```diff
@@ -0,0 +1,82 @@
+import { Box, Group, Text } from '@mantine/core';
+import { Bundle, MedicationKnowledge } from '@medplum/fhirtypes';
+import { AsyncAutocomplete } from '@medplum/react';
+import React, { useCallback } from 'react';
+
+export interface DoseSpotMedicationSelectProps {
+  searchMedications: (term: string) => Promise<Bundle<MedicationKnowledge> | undefined>;
+  onMedicationSelect: (medication: MedicationKnowledge) => void;
+}
+
+export function DoseSpotMedicationSelect({
+  searchMedications,
+  onMedicationSelect,
+}: DoseSpotMedicationSelectProps): React.JSX.Element {
+  const loadOptions = useCallback(
+    async (input: string): Promise<MedicationKnowledge[]> => {
+      if (!input.trim() || input.trim().length < 3) {
+        return [];
+      }
+      const results = await (searchMedications as (term: string) => Promise<Bundle<MedicationKnowledge> | undefined>)(
+        input.trim()
+      );
+
+      if (!results) {
+        return [];
+      }
+
+      return results.entry?.map((entry) => entry.resource as MedicationKnowledge).filter(Boolean) || [];
```

Comment:
unless there's a strong reason not to, I'd suggest rolling all of this unbundling logic into the implementation of the hook (or the bot). Pretty much all clients are going to have to do this. 

That should basically eliminate the need for this wrapper component entirely, allowing you to plug directly into AsyncAutocomplete, as we do here.

https://github.com/medplum/medplum/blob/053b7d09a39edd9026cefac536a40d4462a900b6/examples/medplum-health-gorilla-demo/src/pages/HomePage.tsx#L105-L114